### PR TITLE
Add Support for FMI Bistro „Wilhelm Gastronomie“ and Mensa Garching

### LINKED ISSUE #67: Hunger: Support New Bistro Owner
Apparently the new owner started providing PDF menus, as seen at the MPI page: https://mpi.fs.tum.de/neu-an-der-tum/am-campus/essen-trinken/

The owner's page currently does not provide a link to the bistro PDF. 

So assuming we can automatically retrieve the PDF in the future (which is a high probability), we have two choices:
- Writing another buggy PDF parser
- Displaying the PDF as an image on the page

I think I'd be fine with option 2 (at least as a short term solution). A problem is the landscape format of the PDF which probably sucks on mobile devices since you need to zoom and scroll horizontally...

### PR DESCRIPTION
**Pull request by jpbernius**
Fixes #67

### CODE REVIEW

**Comment by kordianbruck** [comment]
Ok, I'll grant you that this is actually pretty nice. But here is the real deal:

* You are using a library for getting a simple json file (guzzle), while a simple file_get_contents would work just fine in this case. Now don't tell me the error handling is better with guzzle, as you didn't handle any errors in your intermediary lib neither. 
* There is absolutely no caching anywhere - we run to github every time. If githubs down, our service does not work.
* If the api is down, there is currently no error handling or message being printed
* Even though you probably put a lot of effort in it, we cannot just simply start putting every contributors links on the page. While that might not be the biggest issue, you didn't include the url of the api owner neither on your readme or on our page below your link
* You really abstracted the whole thing a bit too much. Come on, a class for each configuration constant? Really? :grimacing:

**Comment by jpbernius** [comment]
> the error handling is better with guzzle

Now added error handling with error messages.

> There is absolutely no caching anywhere

Here you go.

> a class for each configuration constant

There is more to come with this, so this is planning ahead.
This is ready for Mensa as well. ➡️ Added Mensa Garching, so closes #2 as well.

**Comment by jpbernius** [comment]
hunger

File: tpl/hunger.twig [hunk 1]
```diff
@@ -2,23 +2,42 @@
...
+        <h3>FMI Bistro</h3>
+
+        {% if fmiBistroWeek == null %}
+            <p>An error occured while fetching the FMI Bistro menu.</p>
         {% else %}
-            No menu for today 😭 <br/>
-            Maybe check the <a href="{{ linkPage }}">Bistros page</a> and <a href="https://github.com/mammuth/TUM.sexy/issues">open a bug report</a> or redirect all complaints to <a
-                href="mailto:info@wilhelm-gastronomie.de">the Bistro</a>.
+            {% for day in fmiBistroWeek %}
+                <strong>{{day.getDate() | date("l, F j")}}<sup>{{day.getDate() | date("S")}}</sup></strong>
```
Comment (pfent) [inline tpl/hunger.twig]: From a usability perspective: I don't think anyone cares about previous days...

Only showing days >= today would probably be more useful 🎉

Comment (mammuth) [reply tpl/hunger.twig]: Yep, on the old version (with the previous bistro owner) we only showed all future days of the current week 👍

Comment (jpbernius) [reply tpl/hunger.twig]: done

**Comment by kordianbruck** [comment]
@jpbernius does your lib really require php 7.1?

**Comment by jpbernius** [comment]
I was using nullable Types. But they are gone, so I will check to add support for 7.0 as well.

**Comment by jpbernius** [comment]
@kordianbruck I looked into `7.0` compatibility and it would require me to drop use of `void` return type, which I do not see the point of doing with `7.2` being released next week.

**Comment by kordianbruck** [comment]
@jpbernius alright, this will have to wait then. I only have 5.6 in any case. Give me a week to run the upgrades

**Comment by jpbernius** [comment]
Push.

**Comment by mammuth** [comment]
> Push.

@kordianbruck Did you manage to install the needed PHP Version?

**Comment by mammuth** [comment]
> Push.

@kordianbruck Did you manage to install the needed PHP Version?

**Comment by kordianbruck** [comment]
Not Yet. Meanwhile @jpbernius can you remove PHP 7.0 from the travis config so the build runs through?

**Comment by kordianbruck** [comment]
Looks good: http://tum.sexy/hunger/

**Comment by mammuth** [comment]
Awesome!
Thanks for the contribution @jpbernius and thanks for updating the infra @kordianbruck 🎉

**Comment by kordianbruck** [comment]
We are even running PHP 7.2 now :stuck_out_tongue_closed_eyes:

**Pull request merged**